packages/@aws-cdk/cli-lib/lib/cli.ts

```diff
@@ -0,0 +1,237 @@
+import * as core from '@aws-cdk/core';
+import { exec as runCli } from 'aws-cdk/lib';
+import { Synthesizer } from 'aws-cdk/lib/api/cxapp/cloud-executable';
+import { Construct } from 'constructs';
+import { SharedOptions, DeployOptions, DestroyOptions, SynthOptions, ListOptions, StackActivityProgress } from './commands';
+
+/**
+ * AWS CDK CLI operations
+ */
+export interface IAwsCdk {
+
+  /**
+   * cdk deploy
+   */
+  deploy(options: DeployOptions): Promise<void>;
+
+  /**
+   * cdk synth
+   */
+  synth(options: SynthOptions): Promise<void>;
+
+  /**
+   * cdk destroy
+   */
+  destroy(options: DestroyOptions): Promise<void>;
+
+  /**
+   * cdk list
+   */
+  list(options?: ListOptions): Promise<void>;
+}
+
+/**
+ * Options to create an AWS CDK API from a directory
+ */
+export interface AwsCdkCliFromDirectoryProps {
+  /**
+   * the directory the AWS CDK app is in
+   *
+   * @default - current working directory
+   */
+  readonly directory?: string
+
+  /**
+   * command-line for executing your app or a cloud assembly directory
+   * e.g. "node bin/my-app.js"
+   * or
+   * "cdk.out"
+   *
+   * @default - read from cdk.json
+   */
+  readonly app?: string
+}
+
+interface AwsCdkCliProps extends AwsCdkCliFromDirectoryProps {
+  readonly synthesizer?: Synthesizer
```

Comment:
This type should be wrapped with some other (interface) type for the public API surface. Lambdas are not supported by jsii.

Reply:
This is an internal interface. Moving it to private.